packages/kit/src/runtime/app/navigation.js

```diff
@@ -49,3 +50,13 @@ async function prefetchRoutes_(pathnames) {
 
 	await Promise.all(promises);
 }
+
+/**
+ * @returns {Record<string, string>}
+ */
+function routeParams_() {
+	const { routes } = router.parse(new URL(location.href));
+	if (routes.length === 0) return {};
+	const match = routes[0][0].exec(location.pathname);
```

Comment:
This could use a comment as to why this line and the next are `routes[0]`. It seems like it's only checking the first route? Why is that?

Reply:
That's a really good comment. I think if there's multiple ambiguous matching routes it gives them all, we'd probably want the first or the last, I'll need to investigate. The order comes from whatever order they're in generated/manifest.js.

Also I don't love the magic indices after that. I should cast it as a CSRPage type and destructure it instead.

I'll have an update soon.